### PR TITLE
Redesign the "Picture of the day" card

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T261670

### CODE REVIEW

File: app/src/main/res/layout/view_card_featured_image.xml [hunk 1]
```diff
@@ -1,46 +1,85 @@
...
+                    android:text="@string/view_featured_image_card_share"
+                    app:icon="@drawable/ic_share_white_24dp"
+                    app:iconSize="20dp"
+                    app:iconPadding="10dp"
+                    app:iconTint="?attr/colorAccent"/>
 
+            </com.google.android.flexbox.FlexboxLayout>
+        </LinearLayout>
+    </org.wikipedia.views.WikiCardView>
 </LinearLayout>
```
Comment (sharvaniharan) [inline app/src/main/res/layout/view_card_featured_image.xml:85]: can we add the newline please?

Comment (cooltey) [reply app/src/main/res/layout/view_card_featured_image.xml:88]: Sure!

File: app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java [hunk 2]
```diff
@@ -29,43 +31,63 @@
...
         headerView.setCallback(callback);
     }
 
-    private void image(@NonNull Uri uri) {
-        imageView.loadImage(uri);
+    private void image(@NonNull FeaturedImage image) {
+        if (containerView.getWidth() == 0) {
+            // The getWidth() from containerView will return zero in the most beginning one or two cards.
+            // The imageHasLoaded is to avoid keeping setting up the image.
+            containerView.getViewTreeObserver().addOnGlobalLayoutListener(() -> loadImage(image));
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java:67]: Can we just do a `post()` here, and not keep track of whether `imageHasLoaded`?

File: app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java [hunk 3]
```diff
@@ -29,43 +31,63 @@
...
+        } else {
+            loadImage(image);
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java:69]: in fact, can we put it in a `post()` regardless of whether `containerView.getWidth() == 0`?

**Review by dbrant — CHANGES_REQUESTED** [review]
See comments inline. Also, the bottom shadow of the card is cut off:

image

File: app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java (hunk 2 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java]: You are correct! That's a proper way of doing it.

File: app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java (hunk 3 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java]: Done!

**Comment by cooltey** [comment]
> 
> 
> See comments inline. Also, the bottom shadow of the card is cut off:
> 
> image

Thanks and done!

**Pull request merged**